applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormSycnronizationService.cs

```diff
@@ -34,105 +37,145 @@ public class ApplicationFormSycnronizationService :
+                        await _applicationFormVersionAppService.TryInitializeApplicationFormVersion(formId, version, applicationFormIdGuid, formVersionId, published);
+                        FormVersionsInitializedVersionHash.Add(formVersionId);
+                    }
+                    else
+                    {
+                        // Process the form submission.
+                        ApplicationForm applicationForm = ObjectMapper.Map<ApplicationFormDto, ApplicationForm>(applicationFormDto);
+                        var result = await _intakeFormSubmissionManager.ProcessFormSubmissionAsync(applicationForm, submissionData);
+                        AddFact("Synchronizing Data - Form Version: ", $"{version} Unity Submission ID: {result}");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError("ApplicationFormSycnronizationService->SynchronizeFormSubmissions Exception: {ex.Message}", ex.Message);
+            }
+        }
+
+        public async Task<HashSet<string>> GetMissingSubmissions(int numberOfDaysToCheck)
+        {
+            _facts = new List<Fact>();
+
             HashSet<string> missingSubmissions = new HashSet<string>();
             // Get all forms with api keys
-            ApplicationFormDtoList = (List<ApplicationFormDto>?)await GetConnectedApplicationFormsAsync();
-            List<Fact> facts = new ();
+            applicationFormDtoList = (List<ApplicationFormDto>?)await GetConnectedApplicationFormsAsync();
 
-            if (ApplicationFormDtoList != null)
+            if (applicationFormDtoList != null)
             {
-                int numberOfDaysToCheck = PREVIOS_DAY;
-                Logger.LogInformation("Appform SYNC: Iterating Forms ");
-                foreach (ApplicationFormDto applicationFormDto in ApplicationFormDtoList)
+                AddFact("Forms Count: ", "" + applicationFormDtoList.Count);
+                foreach (ApplicationFormDto applicationFormDto in applicationFormDtoList)
                 {
                     try
                     {
                         HashSet<string> newChefsSubmissions = await GetChefsSubmissions(applicationFormDto, numberOfDaysToCheck);
                         HashSet<string> existingSubmissions = GetSubmissionsByForm(applicationFormDto.Id);
                         missingSubmissions = newChefsSubmissions.Except(existingSubmissions).ToHashSet();
-                        Logger.LogInformation("Appform SYNC: Looking up missing: " + missingSubmissions);
+
+                        AddFact("------------------------------------", "----------------------------------------");
+                        AddFact("Application Form Name: ", applicationFormDto.ApplicationFormName);
+                        AddFact("Missing Submissions Count: ", missingSubmissions.Count.ToString());
+
                         if (missingSubmissions != null && missingSubmissions.Count > 0)

```

Comment:
missingSubmissions can never be null, so dont think this check is necessary

Reply:
Removed - fixed